content/_code-samples/tx-serialization/js/index.js

```diff
@@ -0,0 +1,687 @@
+            const postfix = Buffer.alloc(5)
+
+            return Buffer.concat([prefix, codeAscii, postfix]).toString("hex")
+        } else if (HEX_REGEX.test(codeString)) {
+            // raw hex code
+            return Buffer.from(codeString).toString("hex")
+        }
+
+        throw new Error("invalid currency code")
+    }
+
+    /**
+     *    Serializes a hexadecimal string as binary and confirms that it's 128 bits
+     *
+     * @param contents
+     * @returns {string}
+     */
+    hash128ToBytes(contents) {
+        const buffer = this.hashToBytes(contents)
+        if(buffer.length !== 16) {
+            // 16 bytes = 128 bits
+            throw new Error("Hash128 is not 128 bits long")
+        }
+
+        return buffer.toString("hex")
+    }
+
+    /**
+     *    Serializes a hexadecimal string as binary and confirms that it's 160 bits
+     *
+     * @param contents
+     * @returns {string}
+     */
+    hash160ToBytes(contents) {
+        const buffer = this.hashToBytes(contents)
+        if(buffer.length !== 20) {
+            // 20 bytes = 160 bits
+            throw new Error("Hash160 is not 160 bits long")
+        }
+
+        return buffer.toString("hex")
+    }
+
+    /**
+     *    Serializes a hexadecimal string as binary and confirms that it's 128 bits
+     *
+     * @param contents
+     * @returns {string}
+     */
+    hash256ToBytes(contents) {
+        const buffer = this.hashToBytes(contents)
+        if(buffer.length !== 32) {
+            // 32 bytes = 256 bits
+            throw new Error("Hash256 is not 256 bits long")
+        }
+
+        return buffer.toString("hex")
+    }
+
+    /**
+     *     Helper function; serializes a hash value from a hexadecimal string
+     *     of any length.
+     *
+     * @param contents
+     * @returns {string}
+     */
+    hashToBytes(contents) {
+        return Buffer.from(contents).toString("hex")
+    }
+
+    /**
+     *     Serialize an object from decoded JSON.
+     *     Each object must have a type wrapper and an inner object. For example:
+     *
+     *     {
+     *         // type wrapper
+     *         "SignerEntry": {
+     *             // inner object
+     *             "Account": "rUpy3eEg8rqjqfUoLeBnZkscbKbFsKXC3v",
+     *             "SignerWeight": 1
+     *         }
+     *     }
+     *
+     *     Puts the child fields (e.g. Account, SignerWeight) in canonical order
+     *     and appends an object end marker.
+     *
+     * @param object
+     * @returns {string}
+     */
+    objectToBytes(object) {
+        const childOrder = Object.keys(object).sort(sortFuncCanonical.bind(this))
+
+        let fieldsAsBytes = [];
+
+        for (const fieldName of childOrder) {
+            if (this.definitions["FIELDS"][fieldName]["isSerialized"]) {
+                const fieldValue = object[fieldName]
+                const fieldBytes = this.fieldToBytes(fieldName, fieldValue)
+                logger(fieldName + ": " + fieldBytes)
```

Comment:
I'd recommend changing this to skip the `ID Prefix` values when logging in verbose mode if possible because currently looks like the ID Prefix should appear twice, when in reality it only appears once. 

Ex. (Current)
```
Serializing field OfferSequence of type UInt32
IdPrefix is: 2019
OfferSequence: 2019001abed7
```

vs

Recommended:
```
Serializing field OfferSequence of type UInt32
IdPrefix is: 2019
OfferSequence: 001abed7
```

(The number of bytes can be a variable amount of bytes (2-4 is common) - so it may require another helper, or getting more information in the return type of `this.fieldToBytes`)

Reply:
In now handle the logging in fieldToBytes function, output is as expected

Reply:
Done, output now corresponds to what you recommended